lib/internal/mime.js

```diff
@@ -0,0 +1,403 @@
+'use strict';
+
+const {
+  ObjectCreate,
+  RegExpPrototypeExec,
+  RegExpPrototypeSymbolMatch,
+  RegExpPrototypeSymbolMatchAll,
+  RegExpPrototypeSymbolReplace,
+  RegExpPrototypeSymbolSearch,
+  RegExpPrototypeSymbolSplit,
+  StringPrototypeToLowerCase,
+  StringPrototypeSlice,
+  StringPrototypeCharAt,
+  StringPrototypeIndexOf,
+  SymbolIterator,
+  SymbolMatch,
+  SymbolMatchAll,
+  SymbolReplace,
+  SymbolSearch,
+  SymbolSplit,
+  SafeMap: Map,
+} = primordials;
+const {
+  ERR_INVALID_MIME_SYNTAX
+} = require('internal/errors').codes;
+
+function SafeStringPrototypeSearch(str, regexp) {
+  regexp.lastIndex = 0;
+  const match = RegExpPrototypeExec(regexp, str);
+  return match ? match.index : -1;
+}
+
+function hardenRegExp(pattern) {
+  pattern[SymbolMatch] = RegExpPrototypeSymbolMatch;
+  pattern[SymbolMatchAll] = RegExpPrototypeSymbolMatchAll;
+  pattern[SymbolReplace] = RegExpPrototypeSymbolReplace;
+  pattern[SymbolSearch] = RegExpPrototypeSymbolSearch;
+  pattern[SymbolSplit] = RegExpPrototypeSymbolSplit;
+  return pattern;
+}
+
+const NotHTTPTokenCodePoint = hardenRegExp(/[^!#$%&'*+\-.^_`|~A-Za-z0-9]/g);
+const NotHTTPQuotedStringCodePoint = hardenRegExp(/[^\t\u0020-~\u0080-\u00FF]/g);
+
+const END_BEGINNING_WHITESPACE = hardenRegExp(/[^\r\n\t ]|$/);
+const START_ENDING_WHITESPACE = hardenRegExp(/[\r\n\t ]*$/);
+
+function toASCIILower(str) {
+  let ret = '';
+  for (const c of str) {
+    if (c >= 'A' && c <= 'Z') {
+      ret += StringPrototypeToLowerCase(c);
+    } else {
+      ret += c;
+    }
+  }
+  return ret;
```

Comment:
for..of is unsafe.
```suggestion
  return ArrayPrototypeJoin(
    ArrayFrom(
      str,
      (c) => (c >= 'A' && c <= 'Z' ? StringPrototypeToLowerCase(c) : c)
    )
  );
```

Reply:
`ArrayFrom` is as unsafe as for..of, it's relying on `String.prototype[@@iterator]` which is user-mutable. You'd need to use either `SafeStringIterator` or a for loop (the latter is probably the easiest for V8 to optimize).